src/vm/invokeutil.cpp

```diff
@@ -718,9 +722,14 @@ OBJECTREF InvokeUtil::CreateObject(TypeHandle th, void * pValue) {
         {
             // Don't use MethodTable::Box here for perf reasons
             PREFIX_ASSUME(pMT != NULL);
-            obj = AllocateObject(pMT);
             DWORD size = pMT->GetNumInstanceFieldBytes();
-            memcpyNoGCRefs(obj->UnBox(), pValue, size);
+
+            UINT64 capturedValue;
+            memcpyNoGCRefs(&capturedValue, pValue, size); // Must capture the primitive value before we allocate the boxed object which can trigger a GC.
+
+            INDEBUG(pValue = (LPVOID)0xcccccccc); // We're about to allocate a GC object - can no longer trust pValue
+            obj = AllocateObject(pMT);
+            memcpyNoGCRefs(obj->UnBox(), &capturedValue, size);
         }
         break;
```

Comment:
`ELEMENT_TYPE_BYREF` should not be needed anymore. Also, it is referring to resource string that no longer exists.